docs-2.0/3.ngql-guide/7.general-query-statements/2.match.md

```diff
@@ -12,51 +12,191 @@ The syntax of `MATCH` is relatively more flexible compared with that of other qu
+
+> **OpenCypher Compatibility**
+> 
+> For now, nGQL does not support the following operations:
+> 
+> * Scan all vertices and edges with `MATCH`.
+> * Scan all vertices with one tag or all edges of one edge type with `MATCH`.
+
+
+## Use patterns in MATCH statements
+
+This topic uses the `nba` dataset as the sample dataset. Before using `MATCH` in the [match on property](#match-modes) mode, make sure there is at least one index for the property. If there are existing data, rebuild the index to make it take effect on all data.
+
+```nGQL
+nebula> CREATE TAG INDEX name ON player(name(20)); // Create an index on the name property.
+Execution succeeded (time spent 2957/3986 us)
+
+nebula> REBUILD TAG INDEX name; // Rebuild the index.
++------------+
+| New Job Id |
++------------+
+| 121        |
++------------+
+Got 1 rows  (time spent 2676/3990 us)
+
+nebula> SHOW JOB 121; // Make sure the rebuild job succeeded.
++----------------+---------------------+------------+------------+------------+
+| Job Id(TaskId) | Command(Dest)       | Status     | Start Time | Stop Time  |
++----------------+---------------------+------------+------------+------------+
+| 121            | "REBUILD_TAG_INDEX" | "FINISHED" | 1607073046 | 1607073046 |
++----------------+---------------------+------------+------------+------------+
+| 0              | "storaged2"         | "FINISHED" | 1607073046 | 1607073046 |
++----------------+---------------------+------------+------------+------------+
+| 1              | "storaged0"         | "FINISHED" | 1607073046 | 1607073046 |
++----------------+---------------------+------------+------------+------------+
+| 2              | "storaged1"         | "FINISHED" | 1607073046 | 1607073046 |
++----------------+---------------------+------------+------------+------------+
+Got 4 rows (time spent 1186/2998 us)
+```
+
+### Match vertices
+
+You can use a user-defined variable in a pair of parentheses to represent a vertex in a pattern. For example: `(v)`.
+
+Tags are specified with `:<tag_name>` and tag properties are specified with `{<prop_name>: <prop_value>}` after the `:<tag_name>`.
+
+The following example uses the name property to match a vertex.
+
+```nGQL
+nebula> MATCH (v:player{name:"Tim Duncan"}) RETURN v;
++----------------------------------------------------+
+| v                                                  |
++----------------------------------------------------+
+| ("player100" :player{name: "Tim Duncan", age: 42}) |
++----------------------------------------------------+
+```
+
+The WHERE clause can do the same thing:
+
+```nGQL
+nebula> MATCH (v:player) WHERE v.name == "Tim Duncan" RETURN v;
++----------------------------------------------------+
+| v                                                  |
++----------------------------------------------------+
+| ("player100" :player{name: "Tim Duncan", age: 42}) |
++----------------------------------------------------+
+```
+
+> **OpenCypher Compatibility**
```

Comment:
stringid and integer id are a big compatibility issue.

Reply:
Yes, and there is a compatibility note about this.